druid/src/widget/common.rs

```diff
@@ -0,0 +1,85 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{Affine, Point, Size};
+
+// These are based on https://api.flutter.dev/flutter/painting/BoxFit-class.html
+#[derive(Clone, Copy, PartialEq)]
+pub enum FillStrat {
+    /// As large as posible without changing aspect ratio of image and all of image shown
+    Contain,
+    /// As large as posible with no dead space so that some of the image may be clipped
+    Cover,
+    /// Fill the widget with no dead space, aspect ratio of widget is used
+    Fill,
+    /// Fill the hight with the images aspect ratio, some of the image may be clipped
+    FitHeight,
+    /// Fill the width with the images aspect ratio, some of the image may be clipped
+    FitWidth,
+    /// Do not scale
+    None,
+    /// Scale down to fit but do not scale up
+    ScaleDown,
+}
+
+impl Default for FillStrat {
+    fn default() -> Self {
+        FillStrat::Contain
+    }
+}
+
+impl FillStrat {
+    /// Calculate an origin and scale for an image with a given `FillStrat`.
+    ///
+    /// This takes some properties of a widget and a fill strategy and returns an affine matrix
+    /// used to position and scale the image in the widget.
+    pub fn affine_to_fill(&self, parent: Size, fit_box: Size) -> Affine {
+        let scalex = parent.width / fit_box.width;
+        let scaley = parent.height / fit_box.height;
+
+        let scale: Point = match self {
+            FillStrat::Contain => {
+                let scale = scalex.min(scaley);
+                Point { x: scale, y: scale }
+            }
+            FillStrat::Cover => {
+                let scale = scalex.max(scaley);
+                Point { x: scale, y: scale }
+            }
+            FillStrat::Fill => Point {
+                x: scalex,
+                y: scaley,
+            },
+            FillStrat::FitHeight => Point {
+                x: scaley,
+                y: scaley,
+            },
+            FillStrat::FitWidth => Point {
+                x: scalex,
+                y: scalex,
+            },
+            FillStrat::ScaleDown => {
+                let scale = scalex.min(scaley).min(1.0);
+                Point { x: scale, y: scale }
+            }
+            FillStrat::None => Point { x: 1.0, y: 1.0 },
+        };
+
+        let origin_x = (parent.width - (fit_box.width * scale.x)) / 2.0;
+        let origin_y = (parent.height - (fit_box.height * scale.y)) / 2.0;
+        let origin = Point::new(origin_x, origin_y);
```

Comment:
it feels like we can also just delete this line and use `origin_x` and `origin_y` directly.